redbpf/src/cpus.rs

```diff
@@ -10,9 +10,20 @@ use std::io::Error;
 use std::str::FromStr;
 
 const SYS_CPU_ONLINE: &str = "/sys/devices/system/cpu/online";
+const SYS_CPU_POSSIBLE: &str = "/sys/devices/system/cpu/possible";
 
 pub type CpuId = i32;
 
+lazy_static! {
+    // possible cpu is static after kernel boot time. So it's OK to cache the
```

Comment:
What happens if you use some cgroup magic to set cpu affinity? Eg. libvirt can [do this](https://docs.fedoraproject.org/en-US/Fedora/13/html/Virtualization_Guide/ch25s06.html), meaning the host will not schedule tasks on the logical CPU, possibly affecting this too.

Reply:
Oh that's a good point, that I've never considered before.
I agree with you. Performance gain would be little, but perhaps there exist many unknown corner cases the assumption becomes incorrect.
Before removing caching logic, I am going to investigate the situations you mentioned. Thanks,